pkg/integrations/integrations.go

```diff
@@ -66,44 +70,53 @@ func (c VersionedIntegrations) MarshalYAML() (interface{}, error) {
 // IsZero implements yaml.IsZeroer.
 func (c VersionedIntegrations) IsZero() bool {
 	switch {
-	case c.configV1 != nil:
-		return reflect.ValueOf(*c.configV1).IsZero()
-	case c.configV2 != nil:
-		return reflect.ValueOf(*c.configV2).IsZero()
+	case c.ConfigV1 != nil:
+		return reflect.ValueOf(*c.ConfigV1).IsZero()
+	case c.ConfigV2 != nil:
+		return reflect.ValueOf(*c.ConfigV2).IsZero()
 	default:
 		return len(c.raw) == 0
 	}
 }
 
 // ApplyDefaults applies defaults to the subsystem based on globals.
 func (c *VersionedIntegrations) ApplyDefaults(scfg *server.Config, mcfg *metrics.Config) error {
-	if c.version != integrationsVersion2 {
-		return c.configV1.ApplyDefaults(scfg, mcfg)
+	if c.version != IntegrationsVersion2 {
+		return c.ConfigV1.ApplyDefaults(scfg, mcfg)
 	}
-	return c.configV2.ApplyDefaults(mcfg)
+	return c.ConfigV2.ApplyDefaults(mcfg)
 }
 
-// setVersion completes the deferred unmarshal and unmarshals the raw YAML into
+// SetVersion completes the deferred unmarshal and unmarshals the raw YAML into
 // the subsystem config for version v.
-func (c *VersionedIntegrations) setVersion(v integrationsVersion) error {
+func (c *VersionedIntegrations) SetVersion(v integrationsVersion, logger log.Logger) error {
 	c.version = v
 
 	switch c.version {
-	case integrationsVersion1:
+	case IntegrationsVersion1:
 		cfg := v1.DefaultManagerConfig
-		c.configV1 = &cfg
-		return yaml.UnmarshalStrict(c.raw, c.configV1)
-	case integrationsVersion2:
+		c.ConfigV1 = &cfg
+		err := yaml.UnmarshalStrict(c.raw, c.ConfigV1)
+		// Node exporter has some post-processing that has to be done for migrations
+		if err != nil {
+			return err
+		}
+		if c.ConfigV1.Integrations.NodeExporter != nil {
+			return c.ConfigV1.Integrations.NodeExporter.Config.PostProcessing()
```

Comment:
This is one thing that needed to change, instead of doing the node_exporter old -> new field name conversion during unmarshal move it to post processing.